app/lib/router/general_router.dart

```diff
@@ -122,10 +122,12 @@ final generalRoutes = [
     name: Routes.createPin.name,
     path: Routes.createPin.route,
     pageBuilder: (context, state) {
+      final String? spaceId = state.uri.queryParameters['spaceId'];
       return NoTransitionPage(
         key: state.pageKey,
         child: CreatePinPage(
-          initialSelectedSpace: state.uri.queryParameters['spaceId'],
+          initialSelectedSpace:
+              spaceId != null && spaceId.isNotEmpty ? spaceId : null,
```

Comment:
Empty `spaceId` was causing issue in space selector.

Reply:
I do wonder if it is more readible to do:
```suggestion
              (spaceId?.isNotEmpty == true) ? spaceId : null,
```

or not .. remain unsure...